src/main/java/org/springframework/hateoas/UriTemplate.java

```diff
@@ -67,6 +67,7 @@ public UriTemplate(String template) {
 			String[] names = matcher.group(2).split(",");
 
 			for (String name : names) {
+				name = name.endsWith("*")? name.substring(0, name.length() - 1) : name;
```

Comment:
This bit of code doesn't implement the spec found in RFC6570. For example, if I try to expand the exploding operator per the spec:

```
	@Test
	public void discoversMultipleRequestParamWithStar() {
		UriTemplate template = new UriTemplate("/foo{&bar,foobar*}");

		assertVariables(template,
			new TemplateVariable("bar", VariableType.REQUEST_PARAM_CONTINUED),
			new TemplateVariable("foobar", VariableType.REQUEST_PARAM_CONTINUED));

		String foo = template.expand(new HashMap<String, Object>(){{
			put("bar", "barExpanded");
			put("foobar", new HashMap<String, String>(){{
				put("city", "Clarksville");
				put("state", "TN");
			}});
		}}).toString();
		System.out.println(foo);
	}
```
...then I don't see what's expected:

```
/foo?bar=barExpanded&foobar=%7Bcity=Clarksville,%20state=TN%7D
```

Per the spec, it should be `/foo?bar=barExpanded?city=Clarksville&state=TN`. Meaning this doesn't appear to be much of a solution.